.github/workflows/pr-slack-notification.yaml

```diff
@@ -258,20 +260,8 @@ jobs:
 
           echo "Done"
 
-    # 'secrets.GITHUB_TOKEN' is automatically provided by GitHub Actions for each workflow run. We don't need to manually create or manage this token.
-    # Below, the workflow is granting "write" access to the actions scope of the GITHUB_TOKEN. This enables the workflow to be able to delete cache keys.
-    # If the "write" permission does not enable the ability to delete the cache keys, then the repository owner will need to create a Personal Access Token (PAT) with write permissions, add it as a secret, and then specify the name in GH_TOKEN.
-    permissions:
-      actions: write
-
-  slack-notification:
```

Comment:
Previously, Slack notification was a separate job, why it became part of this job now ?

Reply:
@aparnamichael , I am passing values between different steps of the same job `"fetch_all_pull_requests_and_notify_using_condition_and_cron"` using `GITHUB_ENV` (GitHub Environment variable). To pass values between different jobs, I need to use artifacts or `outputs`. However, in this case, I require both outputs and GITHUB_ENV, which would lead to duplication. GITHUB_ENV is needed for passing values between steps within one job, while outputs are required for passing values between jobs. Therefore, I thought it would be better to avoid adding another job and instead keep it as an additional step. The execution of that step will be based on the values from GITHUB_ENV (env.notify, env.notify_closed). If there is no condition to check for the execution, then we can add it as another job, but here that is not applicable.